compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1115,6 +1116,83 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         }
     }
 
+    /// Visit every module reachable from `start_module` and bring any proc and derive macro
+    /// encountered into `self.macro_map` to be used for suggestions.
+    fn lookup_macros_from_module(&mut self) {
```

Comment:
This is so that the case in `tests/ui/macros/missing-derive-4.rs` (`use serde;`, but no `use serde::Serialize;`) is handled.